OpenIGTLinkIOConfig.cmake.in

```diff
@@ -1,23 +1,42 @@
-SET(OpenIGTLink_DIR "@OpenIGTLink_DIR@")
-FIND_PACKAGE(OpenIGTLink REQUIRED NO_MODULE)
+IF(NOT OpenIGTLink_FOUND)
+  SET(OpenIGTLink_DIR "@OpenIGTLink_DIR@")
+  FIND_PACKAGE(OpenIGTLink REQUIRED NO_MODULE)
+ENDIF()
 IF(EXISTS ${OpenIGTLink_USE_FILE})
   INCLUDE(${OpenIGTLink_USE_FILE})
 ENDIF()
 
-INCLUDE("@CMAKE_BINARY_DIR@/OpenIGTLinkIOTargets.cmake")
+IF(@IS_INSTALL@)
+  get_filename_component(OpenIGTLinkIO_CMAKE_PACKAGE_DIR "${CMAKE_CURRENT_LIST_FILE}" DIRECTORY)
+  get_filename_component(OpenIGTLinkIO_BASE_DIR "${OpenIGTLinkIO_CMAKE_PACKAGE_DIR}" DIRECTORY)
+  get_filename_component(OpenIGTLinkIO_LIBRARY_DIR "${OpenIGTLinkIO_BASE_DIR}" DIRECTORY)
+  get_filename_component(OpenIGTLinkIO_BASE_DIR "${OpenIGTLinkIO_LIBRARY_DIR}" DIRECTORY)
+
+  set(OpenIGTLinkIO_BINARY_DIR "${OpenIGTLinkIO_BASE_DIR}/@OpenIGTLinkIO_BINARY_INSTALL@" DIRECTORY)
+  
+  INCLUDE("${OpenIGTLinkIO_CMAKE_PACKAGE_DIR}/OpenIGTLinkIOTargets.cmake")
+
+  SET(OpenIGTLinkIO_INCLUDE_DIRS "${OpenIGTLinkIO_BASE_DIR}/@OpenIGTLinkIO_INCLUDE_INSTALL@")
+  SET(OpenIGTLinkIO_LIBRARY_DIRS ${OpenIGTLinkIO_BINARY_DIR} ${OpenIGTLinkIO_LIBRARY_DIR})
+ELSE()
+  INCLUDE("@CMAKE_BINARY_DIR@/OpenIGTLinkIOTargets.cmake")
+
+  SET(OpenIGTLinkIO_LIBRARY_DIRS "@OpenIGTLinkIO_LIBRARY_DIRS@")
+  SET(OpenIGTLinkIO_INCLUDE_DIRS "@OpenIGTLinkIO_INCLUDE_DIRS@")
+ENDIF()
 
 # Enable support for old style include/linking
 # It is highly recommended to link to target names, thus 
 #    includes/linking are handled automatically
 SET(OpenIGTLinkIO_FOUND 1)
-SET(OpenIGTLinkIO_INCLUDE_DIRS "@OpenIGTLinkIO_INCLUDE_DIRS@")
-SET(OpenIGTLinkIO_LIBRARY_DIRS "@OpenIGTLinkIO_LIBRARY_DIRS@")
 SET(OpenIGTLinkIO_LIBRARIES "@OpenIGTLinkIO_LIBRARIES@")
 
 IF(@IGTLIO_USE_GUI@)
   SET(IGTLIO_QT_VERSION "@IGTLIO_QT_VERSION@")
   IF(IGTLIO_QT_VERSION EQUAL "5")
-    SET(Qt5_DIR "@Qt5_DIR@")
+	IF(NOT DEFINED Qt5_DIR)
+	  SET(Qt5_DIR "@Qt5_DIR@")
```

Comment:
This opens up the possibility of multiple Qt5 versions used.

If you want add logic here, you could check if Qt5_FOUND and/or Qt5_DIR matches what this build of OpenIGTLinkIO was built with, and fire an error if mismatch.